### PR TITLE
edited map.ts

### PR DESCRIPTION
**Pull request by krishnaglodha**
fixed map.ts

### CODE REVIEW

File: src/ui/map.ts [hunk 1]
```diff
@@ -1533,42 +1538,69 @@ export class Map extends Camera {
...
+     * the update will be delayed until the previous style is loaded.
+     *
+     * If `style` is `null`, the current style will be removed.
+     *
+     * @example
+     * map.updateStyle('https://example.com/style.json');
+     * map.updateStyle({ version: 8, sources: { ... }, layers: [ ... ] });
+     * map.updateStyle(null);
+     */
+    _updateStyle(style: StyleSpecification | string | null, options?: StyleSwapOptions & StyleOptions): Map {
```
Comment (HarelM) [inline src/ui/map.ts:1595]: "_" prefixed method are usually private, I would focus on documenting the public ones, those that doesn't start with "_"

File: src/ui/map.ts [hunk 2]
```diff
@@ -140,13 +141,7 @@ export type MapOptions = {
 };
 
 /**
- * An options object for the gesture settings
```
Comment (HarelM) [inline src/ui/map.ts:143]: This info is important, where did you place it?

File: src/ui/map.ts (hunk 1 above)
Comment (krishnaglodha) [reply src/ui/map.ts:1595]: this will be private ?

**Comment by HarelM** [comment]
Thanks for looking into it!
Above the Map xlass there is a huge comment that need to be split and move the comments to the mapoptions object.

File: src/ui/map.ts (hunk 1 above)
Comment (HarelM) [reply src/ui/map.ts:1595]: Yes, _updateStyle is private I believe. There's probably another one that is public somewhere in this long file 😀

**Comment by krishnaglodha** [comment]
> Thanks for looking into it! Above the Map xlass there is a huge comment that need to be split and move the comments to the mapoptions object.

Can you please share more about it

**Comment by HarelM** [comment]
See https://github.com/maplibre/maplibre-gl-js/blob/254d57df96901d2213438da6001af9b2aa40eeb4/src/ui/map.ts#L239L297

You can also see my commit in this branch on the map.ts file starting the work to break this comment.

**Comment by krishnaglodha** [comment]
> See https://github.com/maplibre/maplibre-gl-js/blob/254d57df96901d2213438da6001af9b2aa40eeb4/src/ui/map.ts#L239L297
> 
> You can also see my commit in this branch on the map.ts file starting the work to break this comment.

ahh so you want to split this comments into different positions?

**Comment by HarelM** [comment]
Yes, add they need to be documented in the "right place".

File: src/ui/map.ts [hunk 3]
```diff
@@ -73,6 +74,9 @@ export type MapOptions = {
     interactive?: boolean;
     /**
      * The HTML element in which MapLibre GL JS will render the map, or the element's string `id`. The specified element must have no children.
+     * @param {number} [options.bearingSnap=7] The threshold, measured in degrees, that determines when the map's
```
Comment (HarelM) [inline src/ui/map.ts:77]: this line should be right above `bearingSnap` no above `container`, it also shouldn't have the `{number}` too.
The comments below also needs fixing.

File: src/ui/map.ts [hunk 4]
```diff
@@ -100,6 +104,33 @@ export type MapOptions = {
     minPitch?: number | null;
     /**
      * The maximum pitch of the map (0-85). Values greater than 60 degrees are experimental and may result in rendering issues. If you encounter any, please raise an issue with details in the MapLibre project.
+     * @param {number} [options.clickTolerance=3] The max number of pixels a user can shift the mouse pointer during a click for it to be considered a valid click (as opposed to a mouse drag).
```
Comment (HarelM) [inline src/ui/map.ts:107]: All this part is incorrect, as mentioned above...

File: src/ui/map.ts [hunk 5]
```diff
@@ -140,13 +174,7 @@ export type MapOptions = {
...
 
 /**
- * An options object for the gesture settings
- * @example
- * let options = {
- *   windowsHelpText: "Use Ctrl + scroll to zoom the map",
- *   macHelpText: "Use ⌘ + scroll to zoom the map",
- *   mobileHelpText: "Use two fingers to move the map",
- * }
+ * An options object for the gesture settings.
```
Comment (HarelM) [inline src/ui/map.ts:177]: I mean the example code and the rest of the comment, not just this specific line.

File: src/ui/map.ts [hunk 6]
```diff
@@ -155,13 +183,22 @@ export type GestureOptions = {
 };
 
 // See article here: https://medium.com/terria/typescript-transforming-optional-properties-to-required-properties-that-may-be-undefined-7482cb4e1585
+/**
+ * A complete version of `MapOptions` where all properties are required.
```
Comment (HarelM) [inline src/ui/map.ts:187]: Not sure this class is even exported, so I'm not sure it's worth adding a comment for it, but your call...

File: src/ui/map.ts [hunk 7]
```diff
@@ -171,14 +208,18 @@ type QueryRenderedFeaturesOptions = {
...
 // the default values, but also the valid range
 const defaultMinPitch = 0;
 const defaultMaxPitch = 60;
-
 // use this variable to check maxPitch for validity
 const maxPitchThreshold = 85;
 
+/**
+ * Default options for configuring the MapLibre GL JS map.
+ * @param {boolean} [options.attributionControl=true] If `true`, an {@link AttributionControl} will be added to the map.
```
Comment (HarelM) [inline src/ui/map.ts:219]: This is incorrect too...

File: src/ui/map.ts [hunk 8]
```diff
@@ -418,6 +417,12 @@ export class Map extends Camera {
     /**
      * The map's {@link TwoFingersTouchPitchHandler}, which allows the user to pitch the map with touch gestures.
      * Find more details and examples using `touchPitch` in the {@link TwoFingersTouchPitchHandler} section.
+     * @param {boolean} [options.maplibreLogo=false] If `true`, the MapLibre logo will be shown.
```
Comment (HarelM) [inline src/ui/map.ts:420]: Not sure why this is here...

**Comment by HarelM** [comment]
I've stopped the review in the middle as this was far from the right approach.
Please let me know when the concept is clear and all the JSDocs comments are migrated to TSDocs comments in this file.

The idea is that types written like `@param {type} paramName` should not appear in the comments and only the param name should be there. If there are places with `[options.paramName]` the comment should appear above the `paramName` in the class/type/interface where it is defined and not in the class that uses this type/interface.

Hope this is clearer now.

File: src/ui/map.ts [hunk 9]
```diff
@@ -2664,7 +2736,13 @@ export class Map extends Camera {
         return [width, height];
     }
 
-    _setupContainer() {
+    /**
+     * Sets up the map container by adding necessary classes, creating the canvas element,
+     * and attaching event listeners.
+     * @param {boolean} [options.interactive=true] If `false`, no mouse, touch, or keyboard listeners will be attached to the map, so it will not respond to interaction.
```
Comment (birkskyum) [inline src/ui/map.ts:2742]: remove type

File: src/ui/map.ts [hunk 10]
```diff
@@ -2508,6 +2573,8 @@ export class Map extends Camera {
      * @param {string} [feature.sourceLayer] (optional) *For vector tile sources, `sourceLayer` is required.*
      * @param {Object} state A set of key-value pairs. The values should be valid JSON types.
      *
+     * @returns {void}
```
Comment (birkskyum) [inline src/ui/map.ts:2576]: remove type

File: src/ui/map.ts [hunk 11]
```diff
@@ -2508,6 +2573,8 @@ export class Map extends Camera {
      * @param {string} [feature.sourceLayer] (optional) *For vector tile sources, `sourceLayer` is required.*
      * @param {Object} state A set of key-value pairs. The values should be valid JSON types.
```
Comment (birkskyum) [inline src/ui/map.ts:2574]: remove type

File: src/ui/map.ts [hunk 12]
```diff
@@ -2508,6 +2573,8 @@ export class Map extends Camera {
      * @param {string} [feature.sourceLayer] (optional) *For vector tile sources, `sourceLayer` is required.*
```
Comment (birkskyum) [inline src/ui/map.ts:2573]: remove type

File: src/ui/map.ts [hunk 13]
```diff
@@ -2481,7 +2546,7 @@ export class Map extends Camera {
     /**
      * Returns the value of the light object.
      *
-     * @returns {LightSpecification} light Light properties of the style.
+     * @returns {LightSpecification} Light properties of the style.
```
Comment (birkskyum) [inline src/ui/map.ts:2549]: remove type

File: src/ui/map.ts [hunk 14]
```diff
@@ -2463,16 +2527,17 @@ export class Map extends Camera {
...
 
     /**
-     * Sets the any combination of light values.
+     * Sets any combination of light values.
      *
-     * @param light Light properties to set. Must conform to the [MapLibre Style Specification](https://maplibre.org/maplibre-style-spec/#light).
+     * @param {LightSpecification} light Light properties to set. Must conform to the [MapLibre Style Specification](https://maplibre.org/maplibre-style-spec/#light).
      * @param {StyleSetterOptions} [options] Options object.
      * @param {boolean} [options.validate=true] Whether to check if the filter conforms to the MapLibre Style Specification. Disabling validation is a performance optimization that should only be used if you have previously validated the values you will be passing to this function.
      * @returns {Map} `this`
```
Comment (birkskyum) [inline src/ui/map.ts:2535]: remove type

File: src/ui/map.ts [hunk 15]
```diff
@@ -2463,16 +2527,17 @@ export class Map extends Camera {
     }
 
     /**
-     * Sets the any combination of light values.
+     * Sets any combination of light values.
      *
-     * @param light Light properties to set. Must conform to the [MapLibre Style Specification](https://maplibre.org/maplibre-style-spec/#light).
+     * @param {LightSpecification} light Light properties to set. Must conform to the [MapLibre Style Specification](https://maplibre.org/maplibre-style-spec/#light).
      * @param {StyleSetterOptions} [options] Options object.
```
Comment (birkskyum) [inline src/ui/map.ts:2533]: remove type

File: src/ui/map.ts [hunk 16]
```diff
@@ -2463,16 +2527,17 @@ export class Map extends Camera {
     }
 
     /**
-     * Sets the any combination of light values.
+     * Sets any combination of light values.
      *
-     * @param light Light properties to set. Must conform to the [MapLibre Style Specification](https://maplibre.org/maplibre-style-spec/#light).
+     * @param {LightSpecification} light Light properties to set. Must conform to the [MapLibre Style Specification](https://maplibre.org/maplibre-style-spec/#light).
```
Comment (birkskyum) [inline src/ui/map.ts:2532]: remove type

File: src/ui/map.ts [hunk 17]
```diff
@@ -2427,32 +2490,33 @@ export class Map extends Camera {
...
-    getSprite() {
+    getSprite(): SpriteSpecification | undefined {
         return this.style.getSprite();
     }
 
     /**
      * Sets the value of the style's sprite property.
      *
-     * @param spriteUrl Sprite URL to set.
+     * @param {string | null} spriteUrl The sprite URL to set.
```
Comment (birkskyum) [inline src/ui/map.ts:2511]: remove type

File: src/ui/map.ts [hunk 18]
```diff
@@ -2380,33 +2443,33 @@ export class Map extends Camera {
...
         return this.style.getGlyphsUrl();
     }
 
     /**
      * Adds a sprite to the map's style.
      *
      * @param {string} id The ID of the sprite to add. Must not conflict with existing sprites.
-     * @param {string} url The URL to load the sprite from
+     * @param {string} url The URL to load the sprite from.
      * @param {StyleSetterOptions} [options] Options object.
```
Comment (birkskyum) [inline src/ui/map.ts:2466]: remove type

File: src/ui/map.ts [hunk 19]
```diff
@@ -2380,33 +2443,33 @@ export class Map extends Camera {
...
+    getGlyphs(): string | null {
         return this.style.getGlyphsUrl();
     }
 
     /**
      * Adds a sprite to the map's style.
      *
      * @param {string} id The ID of the sprite to add. Must not conflict with existing sprites.
-     * @param {string} url The URL to load the sprite from
+     * @param {string} url The URL to load the sprite from.
```
Comment (birkskyum) [inline src/ui/map.ts:2465]: remove type

**Pull request closed without merging**